src/modules/validation/validation-module-manager.js

```diff
@@ -7,25 +7,44 @@ class ValidationModuleManager extends BaseModuleManager {
 
     calculateRoot(assertion) {
         if (this.initialized) {
+            if (!assertion) {
+                throw new Error('Calculation failed: Assertion cannot be null or undefined.');
+            }
             return this.getImplementation().module.calculateRoot(assertion);
         }
+        throw new Error('Validation module is not initialized.');
     }
 
     getMerkleProof(assertion, index) {
         if (this.initialized) {
+            if (!assertion) {
+                throw new Error('Get merkle proof failed: Assertion cannot be null or undefined.');
+            }
             return this.getImplementation().module.getMerkleProof(assertion, index);
         }
+        throw new Error('Validation module is not initialized.');
     }
 
     getHashFunctionName(hashFunctionId) {
         if (this.initialized) {
+            if (!hashFunctionId) {
+                throw new Error(
+                    'Getting function name failed: Function ID cannot be null or undefined.',
+                );
+            }
             return this.getImplementation().module.getHashFunctionName(hashFunctionId);
         }
+        throw new Error('Validation module is not initialized.');
     }
 
     async callHashFunction(hashFunctionId, data) {
         if (this.initialized) {
-            return this.getImplementation().module.callHashFunction(hashFunctionId, data);
+            if (!!hashFunctionId || !!data) {
```

Comment:
should it be &&? If hashFunctionId and data are both truthy, we can call the implementation